message_converter/src/converter_node.cpp

```diff
@@ -148,12 +90,13 @@ void SwiftNavRover::posLLHCallback(const libsbp_ros_msgs::MsgPosLlh::ConstPtr &
     int v_accuracy = msg->v_accuracy;
     int tow = msg->tow;
     int fixed_mode = (msg->flags) & (0b00000111);
-    boost::array<double, 9> covariance = {{h_accuracy, 0, 0, 0, h_accuracy, 0, 0, 0, v_accuracy}};
+    SwiftNavRover::positioning_mode_ = fixed_mode;
+    boost::array<double, 9> covariance = {{h_accuracy/1000.00, 0, 0, 0, h_accuracy/1000.00, 0, 0, 0, v_accuracy/1000.00}};
     ros::Time t = ros::Time::now();
```

Comment:
instead of using `now()`; preserve the timestamp of the incoming message.

Reply:
Instead of using `now()`, the node now uses the GPS tow, and using existing functions from `libsbp_ros_msgs` package, the time is converted to ROS time